CONTRIBUTING.md

```diff
@@ -121,6 +121,42 @@ cabal test ouroboros-consensus:test:consensus-test --test-show-details=direct
 Note the second one cannot be used when we want to provide CLI arguments to the
 test-suite.
 
+# Generating documentation and setting up hoogle
+
+Documentation contains some tikz figures that require some preprocessing for
+them to be displayed. In order to do this, use the documentation script:
+
+```bash
+./scripts/docs/haddocks.sh
+```
+
+This will install `latex-svg-image` from [a
+fork](https://github.com/jasagredo/latex-svg/tree/jasagredo/draw-tikz/latex-svg-image),
+and `cabal-docspec` from a binary (do inspect the `Makefile` to see what it
+does), then building the haddocks for your package.
+
+Often times, it is useful to have a `hoogle` server at hand, with the packages
+and its dependencies. For this, one needs to first modify the global cabal
+configuration:
+
+```bash
+cabal user-config update -a "documentation: True
```

Comment:
Isn't it better to set this per-project in a `cabal.project.local`?

Reply:
With latest versions this can be omitted.

Reply:
Although in any case, in previous versions you need this to have the documentation of the dependencies.